src/Controller/PackageController.php

```diff
@@ -986,6 +989,42 @@ public function removeMaintainerAction(Request $req, #[MapEntity] Package $packa
         ]);
     }
 
+
+    #[Route(path: '/packages/{name:package}/transfer/', name: 'transfer_package', requirements: ['name' => Package::PACKAGE_NAME_REGEX])]
+    public function transferPackageAction(Request $req, #[MapEntity] Package $package, #[CurrentUser] User $user, LoggerInterface $logger): RedirectResponse
+    {
+        $this->denyAccessUnlessGranted(PackageActions::TransferPackage->value, $package);
+
+        $form = $this->createTransferPackageForm($package);
+        $form->handleRequest($req);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            try {
+                $newMaintainers = $form->getData()->getMaintainers()->toArray();
+                $result = $this->packageManager->transferPackage($package, $newMaintainers);
+                $this->getEM()->flush();
+
+                if ($result) {
+                    $usernames = array_map(fn (User $user) => $user->getUsername(), $newMaintainers);
+                    $this->addFlash('success', sprintf('Package has been transferred to %s', implode(', ', $usernames)));
+                } else {
+                    $this->addFlash('warning', 'Package maintainers are identical and have not been changed');
+                }
+
+                return $this->redirectToRoute('view_package', ['name' => $package->getName()]);
+            } catch (\Exception $e) {
+                $logger->critical($e->getMessage(), ['exception', $e]);
+                $this->addFlash('error', 'The package could not be transferred.');
+            }
+        } elseif (!$form->isValid()) {
```

Comment:
Unsubmitted form can't be checked for validity, so this need just a bit different nesting of the if statements.